posts/admiral/valuelevel/derive_param_computed.qmd

```diff
@@ -0,0 +1,72 @@
+---
+title: "Derive a new parameter computed from the value of other parameters"
+author:
+  - name: Kangjie Zhang
+
+description: "Use `admiral::derive_param_computed()` like a calculator to derive new parameters/rows based on existing ones"
+date: "2023-05-03"
+categories: [R, admiral, Valuelevel]
+image: "derive_param_computed.jpg"
+---
+
+<!--------------- my typical setup ----------------->
+
+```{r setup, include=FALSE}
+set.seed(8)
+wide <- 136
+narrow <- 76
+options(width = narrow)
+```
+
+<!--------------- post begins here ----------------->
+
+## Introduction
+
+When creating ADaM Basic Data Structure (BDS) datasets, we often encounter deriving a new parameter based on the analysis values (e.g., AVAL) of other parameters.
+
+The `admiral` function `derive_param_computed()` works like a calculator to derive new records without worrying about merging and combining datasets, all you need is a derivation formula, which also improves the readability of the code. 
+
+
+## Example
+
+A valuelevel validation use case, where `derive_param_computed()` is used to validate a derived parameter - `PARAMCD = ADPCYMG` (Actual Dose per Cycle) in ADEX dataset. 
+
+### Derivation text
+Actual Dose per Cycle is derived from the Total Amount of Dose (`PARAMCD=TOTDOSE`) / Number of Cycles (`PARAMCD=NUMCYC`)
+
+### Derivation formula
+
+In this example, `ADEX.AVAL when ADEX.PARAMCD = ADPCYMG` can be derived as:
+
+$$
+AVAL (PARAMCD = ADPCYMG) = \frac{AVAL (PARAMCD = TOTDOSE)}{AVAL (PARAMCD = NUMCYC)}
+$$
+
+``` r
+# read in data
+path_ads <- ""
+adex <- haven::read_sas(paste0(path_ads, "adex.sas7bdat")) %>% 
+  select(USUBJID, PARAM, PARAMCD, AVAL)
+
+# subset input data to PARAMCD = "TOTDOSE", "NUMCYC"
+adex_TOTDOSE_NUMCYC <- adex %>% filter(PARAMCD %in% c("TOTDOSE", "NUMCYC"))
+# admiral::derive_param_computed()
+val <- 
+  derive_param_computed(
+  adex_TOTDOSE_NUMCYC,
+  by_vars = exprs(USUBJID),
+  parameters = c("TOTDOSE", "NUMCYC"),
+  analysis_value = AVAL.TOTDOSE / AVAL.NUMCYC,
+  set_values_to = exprs(
+    PARAMCD = "ADPCYMG",
+    PARAM = "Actual Dose per Cycle (mg)"
+  )
+) 
+
+# compare with primary
+diffdf::diffdf(adex %>% filter(PARAMCD == "ADPCYMG"), 
+               val %>% filter(PARAMCD == "ADPCYMG"), 
+               keys = c("USUBJID", "PARAMCD"))
```

Comment:
We should mention diffdf and what it does in the Introduction

Reply:
Added a brief intro at the last section, let me know if further explanation needed. Thanks for the review!
> For validation purpose, the [diffdf](https://gowerc.github.io/diffdf/reference/diffdf.html) package is used below to mimic SAS proc compare.